### ISSUE TITLE
Vertical scrollbar is disappearing when open a combobox inside a dialog in Safari

### ISSUE THREAD

**Issue by jcgueriaud1** [vaadin/vaadin-dialog-flow#204]
In Vaadin 14.2.3, only on Safari.

When I'm opening the overlay of the combobox, the dialog inner scrollbar is disappearing.

I can reproduce it in Safari. Chrome seems ok.

Here the code to reproduce it:
```

@Route("dialog")
@CssImport("./styles/shared-styles.css")
@CssImport(value="./styles/dialog-overlay.css", themeFor="vaadin-dialog-overlay")
public class DialogView extends VerticalLayout {

    private String text = "Lorem ipsum dolor sit amet consectetur adipisicing elit. Accusantium reprehenderit distinctio pariatur obcaecati!"
            + "Quidem, voluptatum. Inventore aut eum quis repellat labore ratione, maxime odit saepe debitis assumenda harum magni unde?"
            + " Lorem, ipsum dolor sit amet consectetur adipisicing elit. In deleniti repudiandae magni ex alias pariatur, molestiae dolor "
            + "expedita odit iste deserunt est suscipit aut omnis inventore, libero dolore. Asperiores, assumenda. Lorem ipsum dolor sit amet"
            + " consectetur adipisicing elit. Assumenda id laudantium quod deleniti neque! Expedita consequuntur necessitatibus earum tenetur "
            + "odio provident deleniti sed praesentium illum facilis id, corporis, illo ab.";

    public DialogView() {

        Button button = new Button("Click me", e -> {
            Dialog dialog = new Dialog();
            VerticalLayout verticalLayout = new VerticalLayout();
            verticalLayout.setSizeFull();
            verticalLayout.setPadding(false);
            Div header = new Div();
            header.setText("Header");

            Div footer = new Div();
            footer.setText("footer");
            Div content = new Div();
            Div textContent = new Div();
            ComboBox<String> objectComboBox = new ComboBox<>();
            objectComboBox.setItems("item 1","item 2", "item 3");
            content.add(objectComboBox);
            content.add(textContent);
            textContent.setText(text+text+text+text+text+text);
            content.addClassName("content-wrapper");
            verticalLayout.add(header, content, footer);

            dialog.add(verticalLayout);
            dialog.open();
        });
        add(button);
    }

}

```
dialog-overlay.css
```
div[part="overlay"] {
    display: flex;
}
```

shared-styles.css
```
vaadin-dialog-overlay flow-component-renderer > div {
  height: 100%;
}
.content-wrapper {
  overflow: auto;
}
```

vaadin_14_1_8_safari after opening the combobox
<img width="1117" alt="vaadin_14_1_8_safari" src="https://user-images.githubusercontent.com/51313578/86585519-42ca3c80-bf8f-11ea-92ea-2b2bb0d36fa7.png">

vaadin_14_2_3_safari before opening the combobox
<img width="1117" alt="vaadin_14_2_3_safari_before" src="https://user-images.githubusercontent.com/51313578/86585526-43fb6980-bf8f-11ea-978c-ad392205afb3.png">

vaadin_14_2_3_safari after opening the combobox
<img width="1117" alt="vaadin_14_2_3_safari_after" src="https://user-images.githubusercontent.com/51313578/86585524-4362d300-bf8f-11ea-9981-98e74aa06652.png">

**Comment by jcgueriaud1** [comment]
By default (before opening the combobox), resizer-container is not full sized.

After opening the combobox:
```
[part='overlay'][style] .resizer-container {
height: 100%;
width: 100%;
}
```

![resizerContainer](https://user-images.githubusercontent.com/51313578/86589265-02ba8800-bf96-11ea-81b7-485e64f4d0bd.png)

**Comment by DiegoCardoso** [comment]
Confirmed. 

The `[part='overlay'][style] .resizer-container` style only applies after the combo-box (or another component using overlay) opens, because it sets `pointer-events: none` to other overlays (at the `[part="overlay"]` element). And after the combo-box is closed, `pointer-events` is removed, but the `style` is still present (it can't be seen on your screenshot, but I could test it with `$0.hasAttribute('style)`), so the selector still matches. I could restore the scrollbar, by setting `overflow: auto` to `[part="overlay]`, but I am uncertain about other effects it may cause.

**Comment by jcgueriaud1** [comment]
Thank you.

Unfortunately, if I set the `overflow:auto` to `[part="overlay]`, the entire dialog is scrollable. In my case, I want only the content scrollable (not my header/footer).

**Comment by DiegoCardoso** [comment]
Oh, yeah, that's right. Needs more investigation, then.

**Comment by web-padawan** [comment]
Now when I looking into this, I think that `[part='overlay'][style]` is certainly not the correct CSS selector.

We would need an attribute e.g. `has-bounds` set by `_setBounds` method to apply styles properly.

**Comment by web-padawan** [comment]
Created a reduced test case which shows the actual issue: https://codepen.io/web-padawan/pen/MWKZxRP

**Comment by web-padawan** [comment]
The bug appears to be fixed in Safari 14.0 beta, but present in Safari TP 102:

### Safari 14 beta

![safari14](https://user-images.githubusercontent.com/10589913/88062159-1d355980-cb71-11ea-9ef3-13eb8437aa51.png)

### Safari TP 102

![safaritp](https://user-images.githubusercontent.com/10589913/88062177-26bec180-cb71-11ea-8387-c1362e7d096f.png)

This appears to be a rendering issue to me. I'm not really sure if that's the case, but the only potentially relevant change that I found in the [Technology Preview release notes](https://developer.apple.com/safari/technology-preview/release-notes/) seems to be [this one](https://bugs.webkit.org/show_bug.cgi?id=212419) which landed in TP 108.

I will try to come up with a workaround.
